index.tsx

```diff
@@ -0,0 +1,216 @@
+
+      return () => {
+        parent.removeChild(node)
+        //@ts-ignore updateFlatGraph missing in zdog types
+        parent.updateFlatGraph()
+        state.current.illu.updateGraph()
+      }
+    }
+  }, [parent])
+  return [<parentContext.Provider value={node} children={children} />, node]
+}
+
+export type IllustrationProps = Omit<Zdog.IllustrationOptions, 'element' | 'addTo'> &
+  PropsWithChildren<{
+    style?: CSSProperties
+    element?: 'svg' | 'canvas'
+  }>
+
+const Illustration = React.memo<IllustrationProps>(
+  ({ children, style, resize, element: Element = 'svg', dragRotate, ...rest }) => {
+    const canvas = useRef<HTMLCanvasElement & SVGSVGElement>()
+    const [bind, size] = useMeasure<HTMLDivElement>()
+    const [result, scene] = useZdogPrimitive(Zdog.Anchor, children)
+
+    const state: MutableRefObject<ZdogState> = useRef<ZdogState>({
+      scene,
+      illu: undefined,
+      size: null,
+      subscribers: [],
+      subscribe: fn => {
+        state.current.subscribers.push(fn)
+        return () => (state.current.subscribers = state.current.subscribers.filter(s => s !== fn))
+      },
+    })
+
+    useEffect(() => {
+      state.current.size = size
+      if (state.current.illu) state.current.illu.setSize(size.width, size.height)
+    }, [size])
+
+    useEffect(() => {
+      state.current.illu = new Zdog.Illustration({ element: canvas.current, dragRotate, ...rest })
+      state.current.illu.addChild(scene)
+      state.current.illu.updateGraph()
+
+      let frame: number
+      let active = true
+      const render: FrameRequestCallback = t => {
+        const { size, subscribers } = state.current
+        if (size && size.width && size.height) {
+          // Run global effects
+          globalEffects.forEach(fn => fn(t))
+          // Run local effects
+          subscribers.forEach(fn => fn(t))
+          // Render scene
+          state.current.illu.updateRenderGraph()
+        }
+        if (active) frame = requestAnimationFrame(render)
+      }
+
+      // Start render loop
+      render(0)
+
+      return () => {
+        // Take no chances, the loop has got to stop if the component unmounts
+        active = false
+        cancelAnimationFrame(frame)
+      }
+    }, [])
+
+    // Takes care of updating the main illustration
+    useLayoutEffect(() => void (state.current.illu && applyProps(state.current.illu, rest)), [rest])
+
+    return (
+      <div
+        ref={bind.ref}
+        {...rest}
+        style={{ position: 'relative', width: '100%', height: '100%', overflow: 'hidden', ...style }}>
+        <Element ref={canvas} style={{ display: 'block' }} width={size.width} height={size.height} />
+        {state.current.illu && <stateContext.Provider value={state} children={result} />}
+      </div>
+    )
+  }
+)
+type ZdogComponent<Primitive extends typeof Zdog.Anchor> = ForwardRefExoticComponent<
+  Omit<PrimitiveProps<Primitive>, 'addTo'> & RefAttributes<InstanceType<Primitive>>
```

Comment:
Unsure if I should omit `addTo` here. Is this prop useful/used when using `react-zdog`?